### PR TITLE
Model cloud features explicitly

### PR DESCRIPTION
**Pull request by mpolden**
Hold merge as host-admin must be changed to generate the correct config.

@bratseth

### CODE REVIEW

**Comment by mpolden** [comment]
FYI @freva

File: config-provisioning/src/main/java/com/yahoo/config/provision/Zone.java [hunk 1]
```diff
@@ -38,29 +41,29 @@ public Zone(Environment environment, RegionName region) {
...
-        this.cloudName = cloudName;
+        this.cloud = cloud;
         this.systemName = systemName;
         this.environment = environment;
         this.region = region;
         this.nodeFlavors = Optional.ofNullable(nodeFlavors);
     }
 
     /** Returns the current cloud */
-    public CloudName cloud() { return cloudName; }
```
Comment (bratseth) [inline config-provisioning/src/main/java/com/yahoo/config/provision/Zone.java:63]: Won't this break old model versions?

**Review by bratseth — COMMENTED** [review]
Otherwise, great!

File: config-provisioning/src/main/java/com/yahoo/config/provision/Zone.java (hunk 1 above)
Comment (mpolden) [reply config-provisioning/src/main/java/com/yahoo/config/provision/Zone.java:63]: Yes, thanks!

**Review by bratseth — APPROVED** [review]

**Pull request merged**